--- LuaJIT-2.0.4/src/luaconf.h.orig	2015-05-14 21:30:00.000000000 +0300
+++ LuaJIT-2.0.4/src/luaconf.h	2016-05-31 16:12:39.830117429 +0300
@@ -35,15 +35,15 @@
 #ifndef LUA_LMULTILIB
 #define LUA_LMULTILIB	"lib"
 #endif
-#define LUA_LROOT	"/usr/local"
-#define LUA_LUADIR	"/lua/5.1/"
+#define LUA_LROOT	"/usr"
+#define LUA_LUADIR	"/lua/"
 #define LUA_LJDIR	"/luajit-2.0.4/"
 
 #ifdef LUA_ROOT
 #define LUA_JROOT	LUA_ROOT
 #define LUA_RLDIR	LUA_ROOT "/share" LUA_LUADIR
 #define LUA_RCDIR	LUA_ROOT "/" LUA_MULTILIB LUA_LUADIR
-#define LUA_RLPATH	";" LUA_RLDIR "?.lua;" LUA_RLDIR "?/init.lua"
+#define LUA_RLPATH	";" LUA_RLDIR "?.luac;" LUA_RLDIR "?.lua;" LUA_RLDIR "?/init.lua"
 #define LUA_RCPATH	";" LUA_RCDIR "?.so"
 #else
 #define LUA_JROOT	LUA_LROOT
@@ -54,11 +54,11 @@
 #define LUA_JPATH	";" LUA_JROOT "/share" LUA_LJDIR "?.lua"
 #define LUA_LLDIR	LUA_LROOT "/share" LUA_LUADIR
 #define LUA_LCDIR	LUA_LROOT "/" LUA_LMULTILIB LUA_LUADIR
-#define LUA_LLPATH	";" LUA_LLDIR "?.lua;" LUA_LLDIR "?/init.lua"
+#define LUA_LLPATH	";" LUA_LLDIR "?.luac;" LUA_LLDIR "?.lua;" LUA_LLDIR "?/init.lua"
 #define LUA_LCPATH1	";" LUA_LCDIR "?.so"
 #define LUA_LCPATH2	";" LUA_LCDIR "loadall.so"
 
-#define LUA_PATH_DEFAULT	"./?.lua" LUA_JPATH LUA_LLPATH LUA_RLPATH
+#define LUA_PATH_DEFAULT	"./?.luac;./?.lua" LUA_JPATH LUA_LLPATH LUA_RLPATH
 #define LUA_CPATH_DEFAULT	"./?.so" LUA_LCPATH1 LUA_RCPATH LUA_LCPATH2
 #endif
 
