--- LuaJIT-2.0.4/src/lib_io.c.orig	2015-05-14 21:30:00.000000000 +0300
+++ LuaJIT-2.0.4/src/lib_io.c	2016-06-03 15:03:53.382976002 +0300
@@ -434,6 +434,12 @@ LJLIB_CF(io_tmpfile)
   return iof->fp != NULL ? 1 : luaL_fileresult(L, 0, NULL);
 }
 
+LJLIB_CF(io_fileno)
+{
+  FILE *f = *(FILE**) luaL_checkudata(L, 1, LUA_FILEHANDLE);
+  return (lua_pushinteger(L, fileno(f)), 1);
+}
+
 LJLIB_CF(io_close)
 {
   return lj_cf_io_method_close(L);
